### DISCUSSION TITLE
Is that even possible? Remove syntax noise by compiling.

### DISCUSSION THREAD

**Discussion by zhangenming** [pmndrs/jotai#2608] [Ideas]
<img width="891" alt="c43845dddc8f7b7491a6e505763c749" src="https://github.com/pmndrs/jotai/assets/21235555/b1c8b631-3cdb-4770-9d22-8b1a05495112">

**Comment by zhangenming** [comment]
or...a runtime solution, maybe?
I think it's less noise and more friendly to the development experience
![image](https://github.com/pmndrs/jotai/assets/21235555/3a44f4af-614c-405d-b3eb-9ef62ef6cc0a)

**Comment by dai-shi** [comment]
It might be possible with a compiler, but Jotai's design choice is a runtime solution.
